DigitalLearningSolutions.Data/Services/TrackerActionService.cs

```diff
@@ -51,5 +66,47 @@ public TrackerActionService(ITutorialContentDataService tutorialContentDataServi
 
             return ccObjectives.Any() ? new TrackerObjectiveArrayCc(ccObjectives) : null;
         }
+
+        public TrackerEndpointResponse StoreDiagnosticJson(int? progressId, string? diagnosticOutcome)
+        {
+            if (!progressId.HasValue || string.IsNullOrEmpty(diagnosticOutcome))
+            {
+                return TrackerEndpointResponse.StoreDiagnosticScoreException;
+            }
+
+            try
+            {
+                var diagnosticOutcomes = JsonConvert.DeserializeObject<IEnumerable<DiagnosticOutcome>>(
+                    diagnosticOutcome,
+                    new JsonSerializerSettings
+                    {
+                        MissingMemberHandling = MissingMemberHandling.Error,
+                    }
+                );
+
+                foreach (var diagOutcome in diagnosticOutcomes)
+                {
+                    if (diagOutcome.TutorialId == 0)
+                    {
+                        throw new Exception();

```

Comment:
I think it would be good to include a message in the exception. From a logging perspective it will be a bit clearer as to why we have a failure.